Bowling/Bowling/FrameWithBonusRollFactory.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BowlingApp
+{
+    public class FrameWithBonusRollFactory
+    {
+        public const int totalPinsCount = 10;
+        
+        public Frame Create(string bowlingMarks, int placeCount)
+        {
+            var rollsRemaining = bowlingMarks.Length - placeCount;
+            var roll1 = bowlingMarks.Substring(placeCount, 1);
+            var roll2 = bowlingMarks.Substring(placeCount + 1, 1);
+            var roll3 = "0";
+
+            if (rollsRemaining == 3)
+            {
+                roll3 = bowlingMarks.Substring(placeCount + 2, 1);
+            }
+
+            return ScoreRollsWithBonus(roll1, roll2, roll3);
+            
+        }
+
+        private Frame ScoreRollsWithBonus(string roll1, string roll2, string roll3)
+        {
+            var scoreRoll1 = 0;
```

Comment:
No need to initialize this to `0`, you can simply say: `var scoreRoll1  BonusScoreCalculation(roll1);`